### PR TITLE
tmuxPlugins.tmux-nova: init at 1.2.0

### PR DESCRIPTION
**Pull request by o0th**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Nydragon** [comment]
Hey, thanks for contributing!

Could you please add yourself to the [maintainers](https://github.com/o0th/nixpkgs/blob/add-tmux-nova/maintainers/maintainer-list.nix) file? Check out the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md)

```
       error: undefined variable 'o0th'

          835|       platforms = platforms.unix;
          836|       maintainers = with maintainers; [ o0th ];
             |                                         ^
          837|     };
```

And also squash both of your commits and rename it and the merge request to `tmuxPlugins.tmux-nova: init at 1.2.0" ([Convention](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md))

File: pkgs/misc/tmux-plugins/default.nix [hunk 1]
```diff
@@ -818,4 +818,22 @@ in rec {
...
+      repo = "tmux-nova";
+      rev = "v${version}";
+      sha256 = "16llz3nlyw88lyd8mmj27i0ncyhpfjj5c1yikngf7nxcqsbjmcnh";
+    };
+    meta = with lib; {
+      homepage = "https://github.com/o0th/tmux-nova";
+      description = "tmux-nova theme";
+      license = licenses.mit;
+      platforms = platforms.unix;
+      maintainers = with maintainers; [ o0th ];
```
Comment (Nydragon) [inline pkgs/misc/tmux-plugins/default.nix:836]: Please add yourself to the [maintainers](https://github.com/o0th/nixpkgs/blob/add-tmux-nova/maintainers/maintainer-list.nix) file? Check out the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md)

```
       error: undefined variable 'o0th'

          835|       platforms = platforms.unix;
          836|       maintainers = with maintainers; [ o0th ];
             |                                         ^
          837|     };
```

**Review by nydragon — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -15197,6 +15197,12 @@
     githubId = 30825096;
     name = "Ning Zhang";
   };
+  o0th = {
+    email = "o0th@pm.me";
+    name = "Sabato Luca Guadagno";
+    github = "o0th";
+    githubId = 22490354;
+  };
```
Comment (Nydragon) [inline maintainers/maintainer-list.nix:15205]: I forgot to mention that this needs to be a separate commit: `maintainers: add <handle>`

**Review by nydragon — APPROVED** [review]

**Comment by Nydragon** [comment]
Result of `nixpkgs-review pr 316457` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tmuxPlugins.tmux-nova</li>
  </ul>
</details>

File: pkgs/misc/tmux-plugins/default.nix [hunk 3]
```diff
@@ -818,4 +818,22 @@ in rec {
       sha256 = "hRvkBf+YrWycecnDixAsD4CAHg3KsioomfJ/nLl5Zgs=";
     };
   };
+
+  tmux-nova = mkTmuxPlugin rec {
+    pluginName = "tmux-nova";
```
Comment (Nydragon) [inline pkgs/misc/tmux-plugins/default.nix:823]: It seems like nixpkgs looks for the file named after your package, i.e `tmux_nova.tmux`. 
You can specify a different file by setting:
```suggestion
    pluginName = "tmux-nova";
    rtpFilePath = "nova.tmux";
```

File: pkgs/misc/tmux-plugins/default.nix
Comment (Nydragon) [inline pkgs/misc/tmux-plugins/default.nix:1]: I did not manage to get the theme working. [onedark-theme](https://github.com/odedlaz/tmux-onedark-theme) works properly, so maybe you could check what they are doing. ([nixpkgs plugin](https://github.com/o0th/nixpkgs/blob/bac7a262f535aba8e4f79bea193ec1230422da9b/pkgs/misc/tmux-plugins/default.nix#L459))

The configuration I used for testing: 
<details><summary>Details</summary>
<p>

```nix
{ pkgs, ... }:
{
  boot.loader.systemd-boot.enable = true;
  boot.loader.efi.canTouchEfiVariables = true;

  services.xserver.enable = true;
  programs.nix-ld.enable = true;
  services.xserver.desktopManager.xfce.enable = true;
  users.users.alice = {
    isNormalUser = true;
    extraGroups = [ "wheel" ];
    initialPassword = "test";
  };

  environment.systemPackages = with pkgs; [ git ];

  programs.tmux = {
    enable = true;
    plugins = with pkgs.tmuxPlugins; [ tmux-nova ];
    extraConfig = ''
      set -g @plugin 'o0th/tmux-nova'

      set -g @nova-nerdfonts true
      set -g @nova-nerdfonts-left 
      set -g @nova-nerdfonts-right 

      set -g @nova-segment-mode "#{?client_prefix,Ω,ω}"
      set -g @nova-segment-mode-colors "#50fa7b #282a36"

      set -g @nova-segment-whoami "#(whoami)@#h"
      set -g @nova-segment-whoami-colors "#50fa7b #282a36"

      set -g @nova-pane "#I#{?pane_in_mode,  #{pane_mode},}  #W"

      set -g @nova-rows 0
      set -g @nova-segments-0-left "mode"
      set -g @nova-segments-0-right "whoami"
    '';
  };

  services.xserver.xkb = {
    layout = "fr";
    variant = "";
  };
  fonts = with pkgs; {
    packages = [ (nerdfonts.override { fonts = [ "Noto" ]; }) ];
  };
  system.stateVersion = "23.11";
}
```

</p>
</details> 

@NovaViper could you please check this out and tell me if I missed something?

**Review by nydragon — CHANGES_REQUESTED** [review]

File: pkgs/misc/tmux-plugins/default.nix
Comment (Nydragon) [reply pkgs/misc/tmux-plugins/default.nix:1]: This is added to /etc/tmux.conf:
<details><summary>Details</summary>
<p>

```tmux
# Run plugins
run-shell /nix/store/d795yzckjb10zygk1f99zxs7hsy2cb5w-tmuxplugin-tmux-nova-1.2.0/share/tmux-plugins/tmux-nova/nova.tmux

set -g @plugin 'o0th/tmux-nova'

set -g @nova-nerdfonts true
set -g @nova-nerdfonts-left 
set -g @nova-nerdfonts-right 

set -g @nova-segment-mode "#{?client_prefix,Ω,ω}"
set -g @nova-segment-mode-colors "#50fa7b #282a36"

set -g @nova-segment-whoami "#(whoami)@#h"
set -g @nova-segment-whoami-colors "#50fa7b #282a36"

set -g @nova-pane "#I#{?pane_in_mode,  #{pane_mode},}  #W"

set -g @nova-rows 0
set -g @nova-segments-0-left "mode"
set -g @nova-segments-0-right "whoami"
```

</p>
</details> 

And the plugin does work when I execute `tmux run-shell /nix/store/d795yzckjb10zygk1f99zxs7hsy2cb5w-tmuxplugin-tmux-nova-1.2.0/share/tmux-plugins/tmux-nova/nova.tmux` manually

Comment (NovaViper) [reply pkgs/misc/tmux-plugins/default.nix:1]: Can confirm I'm having issues with it configuring too @Nydragon. Ran `tmux run-shell` for my config and it displays the settings properly, here's my settings (I use Home-Manager to configure tmux)
```
programs.tmux = {
      enable = true;
 ..... many more settings not included
      plugins = with pkgs.tmuxPlugins; [
        sensible
        {
          plugin = yank;
          extraConfig = ''
            # Enable Mouse support for tmux-yank
            set -g @yank_with_mouse on
          '';
        }
        open
        fuzzback
        extrakto
        (mkIf (config.programs.fzf.enable) tmux-fzf)
        {
          plugin = nova;
          extraConfig = ''
            # -- statusline -----------------------------------------------------------------
            set -g @nova-nerdfonts true
            set -g @nova-nerdfonts-left 
            set -g @nova-nerdfonts-right 
            set -g @nova-rows 0
            # Colors
            set -gw window-status-current-style bold
            set -g "@nova-status-style-bg" "${c.base02}"
            set -g "@nova-status-style-fg" "${c.base05}"
            set -g "@nova-status-style-active-bg" "${c.base0D}"
            set -g "@nova-status-style-active-fg" "${c.base00}"
            #set -g "@nova-pane-active-border-style" "${c.base09}"
            #set -g "@nova-pane-border-style" "${c.base00}"
            # Declare Status
            #set -g @nova-segment-prefix "#{?client_prefix,Ω,ω}"
            # From https://github.com/o0th/tmux-nova/issues/33#issuecomment-1509927098
            set -g @nova-segment-action-prefix "#{?client_prefix,PREFIX,}#{?#{==:#{pane_mode},copy-mode}, COPY,}#{?#{==:#{pane_mode},view-mode}, VIEW,󰻀}"
            set -g @nova-segment-action-prefix-colors "${c.base0F} ${c.base00}"
            set -g @nova-pane "#I [#W]"
            #set -g @nova-segment-whoami "#(whoami)@#h"
            #set -g @nova-segment-whoami-colors "$light_purple $white"
            set -g @nova-pane "#I#{?pane_in_mode, #{pane_mode},} [#W]"
            # Declare system info
            set -g @nova-segment-battery "#{battery_icon_status} #{battery_percentage}"
            set -g @nova-segment-battery-colors "${c.base0E} ${c.base00}"
            #set -g @batt_icon_status_charging '↑'
            #set -g @batt_icon_status_discharging '↓'
            set -g @nova-segment-cpu "CPU #{cpu_percentage}"
            set -g @nova-segment-cpu-colors "${c.base09} ${c.base00}"
            set -g @nova-segment-ram "RAM #{ram_percentage}"
            set -g @nova-segment-ram-colors "${c.base0F} ${c.base00}"
            set -g @nova-segment-time "#(date +'%a %m/%d ${
              if cfg.clock24
              then "%H:%M"
              else "%I:%M %p"
            }') #(date +%Z)"
            set -g @nova-segment-time-colors "${c.base03} ${c.base05}"
            set -g @nova-segments-0-left "action-prefix"
            set -g @nova-segments-0-right "battery cpu ram time"
          '';
        }
      ];
    };
```

Comment (NovaViper) [reply pkgs/misc/tmux-plugins/default.nix:1]: Huh I just made an interesting discovery, it looks like it's speciifcally the v1.2.0 version of the plugin that's bugged out. I went back to the 1.1.0 version ([which is what my flake includes](https://github.com/NovaViper/NixConfig/blob/main/pkgs/tmux-plugins/nova/default.nix)), and it executes the settings correctly and displays.

Comment (o0th) [reply pkgs/misc/tmux-plugins/default.nix:1]: I did add `rtpFilePath` maybe now is working

Comment (Nydragon) [reply pkgs/misc/tmux-plugins/default.nix:1]: The right file is now being set, but the theme does not start automatically.

<details><summary>On tmux start:</summary>
<p>

image

</p>
</details> 

<details><summary>When running nova.tmux manually</summary>
<p>

image

</p>
</details>

Comment (Nydragon) [reply pkgs/misc/tmux-plugins/default.nix:1]: > Huh I just made an interesting discovery, it looks like it's speciifcally the v1.2.0 version of the plugin that's bugged out. I went back to the 1.1.0 version ([which is what my flake includes](https://github.com/NovaViper/NixConfig/blob/main/pkgs/tmux-plugins/nova/default.nix)), and it executes the settings correctly and displays.

According to Git there is [no difference between v1.2.0 and 1.10](https://github.com/o0th/tmux-nova/compare/v1.2.0...1.1.0). What is the commit hash you are on ?

Comment (Nydragon) [reply pkgs/misc/tmux-plugins/default.nix:1]: I think it could be that nova is unable to find a nerdfont when being run by tmux:
https://github.com/o0th/tmux-nova/compare/1.1.0...v1.2.0#diff-bda9845cafb8a2caec42c6dd1b744216c9c3b2160bb69004f50ec251c22382e4R96

```
    if [ $nerdfonts = true ] && [ $first_left_segment = true ]; then
      tmux set-option -ga status-left "#[fg=${segment_colors[0]}#,bg=${segment_colors[1]}]"
      tmux set-option -ga status-left "$first_left_nerdfont"
    fi
```

It would be necessary to bring nerdfont into the scope of the plugin

Comment (Nydragon) [reply pkgs/misc/tmux-plugins/default.nix:1]: I got it to work, the config settings need to be put into [extraConfigBeforePlugins](https://mynixos.com/nixpkgs/option/programs.tmux.extraConfigBeforePlugins)

**Review by nydragon — APPROVED** [review]

File: pkgs/misc/tmux-plugins/default.nix
Comment (Nydragon) [reply pkgs/misc/tmux-plugins/default.nix:1]: > Huh I just made an interesting discovery, it looks like it's speciifcally the v1.2.0 version of the plugin that's bugged out. I went back to the 1.1.0 version ([which is what my flake includes](https://github.com/NovaViper/NixConfig/blob/main/pkgs/tmux-plugins/nova/default.nix)), and it executes the settings correctly and displays.

Could you try setting the nova options in [extraConfigBeforePlugins](https://mynixos.com/nixpkgs/option/programs.tmux.extraConfigBeforePlugins) ? Might be the fix

Comment (NovaViper) [reply pkgs/misc/tmux-plugins/default.nix:1]: > > Huh I just made an interesting discovery, it looks like it's speciifcally the v1.2.0 version of the plugin that's bugged out. I went back to the 1.1.0 version ([which is what my flake includes](https://github.com/NovaViper/NixConfig/blob/main/pkgs/tmux-plugins/nova/default.nix)), and it executes the settings correctly and displays.
> 
> Could you try setting the nova options in [extraConfigBeforePlugins](https://mynixos.com/nixpkgs/option/programs.tmux.extraConfigBeforePlugins) ? Might be the fix

👀 I hope that doesn't mean it's completely incompatible with Home-Manager, I got alot of my settings confused via Home-Manager and having to switch all of it over would actually break much of my Home-Manager configurations altogether.

Comment (NovaViper) [reply pkgs/misc/tmux-plugins/default.nix:1]: Also putting it in extraConfigBeforePlugins (with the home-manager tmux disabled) doesn't make it run either 👀

Comment (Nydragon) [reply pkgs/misc/tmux-plugins/default.nix:1]: > > > Huh I just made an interesting discovery, it looks like it's speciifcally the v1.2.0 version of the plugin that's bugged out. I went back to the 1.1.0 version ([which is what my flake includes](https://github.com/NovaViper/NixConfig/blob/main/pkgs/tmux-plugins/nova/default.nix)), and it executes the settings correctly and displays.
> > 
> > 
> > Could you try setting the nova options in [extraConfigBeforePlugins](https://mynixos.com/nixpkgs/option/programs.tmux.extraConfigBeforePlugins) ? Might be the fix
> 
> 👀 I hope that doesn't mean it's completely incompatible with Home-Manager, I got alot of my settings confused via Home-Manager and having to switch all of it over would actually break much of my Home-Manager configurations altogether.

I'd hope that the extraConfig in HM's programs.tmux.plugins is executed before running the plugin... You could check that by looking in your .tmux.conf

And you could review the changes & approve, might make someone with write access more likely to check out and approve this pr

Comment (NovaViper) [reply pkgs/misc/tmux-plugins/default.nix:1]: > I'd hope that the extraConfig in HM's programs.tmux.plugins is executed before running the plugin... You could check that by looking in your .tmux.conf

@Nydragon Ah yeah, I did take a look at my tmux config from HM and found that it does actually put the extra configs before actually running the plugins; so that isn't the issue there.


> And you could review the changes & approve, might make someone with write access more likely to check out and approve this pr

Hey how do I do that exactly? Ngl I haven't really used this feature of Github before 😅

Comment (Nydragon) [reply pkgs/misc/tmux-plugins/default.nix:1]: I think you should have received a request for review just now, on the top of this page there is a button that will bring you to the page where you can approve changes.

IDK if adding a second approval would really change much, but worth a shot

**Review by NovaViper — APPROVED** [review]

**Comment by drupol** [comment]
Congrats on your first PR, keep them coming!

File: pkgs/misc/tmux-plugins/default.nix [hunk 4]
```diff
@@ -818,4 +818,23 @@ in rec {
...
+    rtpFilePath = "nova.tmux";
+    version = "1.2.0";
+    src = fetchFromGitHub {
+      owner = "o0th";
+      repo = "tmux-nova";
+      rev = "v${version}";
+      sha256 = "16llz3nlyw88lyd8mmj27i0ncyhpfjj5c1yikngf7nxcqsbjmcnh";
```
Comment (drupol) [inline pkgs/misc/tmux-plugins/default.nix:830]: For next time, use `hash` instead.
